subprojects/docs/src/docs/userguide/groovy_plugin.adoc

```diff
@@ -253,6 +253,39 @@ include::sample[dir="groovy/compilationAvoidance/groovy", files="build.gradle[ta
 include::sample[dir="groovy/compilationAvoidance/kotlin", files="build.gradle.kts[tags=groovy-compilation-avoidance]"]
 ====
 
+[[sec:incremental_groovy_compile]]
+== Incremental Groovy compilation
+
+Since 5.6, Gradle introduces an experimental incremental Groovy compiler. To enable incremental compilation for Groovy, you need:
+
+* Enable <<groovy_plugin.adoc#sec:groovy_compilation_avoidance,Groovy compilation avoidance>>.
+* Explicitly enable incremental Groovy compilation in the build script:
+
+.Enable Groovy compilation
+====
+include::sample[dir="groovy/incrementalCompilation/groovy",files="build.gradle[tags=enable-groovy-incremental]"]
+include::sample[dir="groovy/incrementalCompilation/kotlin",files="build.gradle.kts[tags=enable-groovy-incremental]"]
+====
+
+This gives you the following benefits:
+
+* Incremental builds are much faster.
+* If only a small sets of Groovy source files are changed, only the affected source files will be recompiled.
+  Classes that don't need to be recompiled remain unchanged in the output directory.
+  For example, if you only change a few Groovy test classes, you don't need to recompile all Groovy test source files - only the changed ones need to be recompiled.
+
+To understand how incremental compilation works, see <<java_plugin.adoc#sec:incremental_compile,Incremental Java compilation>> for a detailed overview.
```

Comment:
Why is the `I` capitalized in `Incremental`, though `compilation` is not? Should we consistently use `incremental Java/Groovy compilation` in the middle of a sentence?

Reply:
Yeah, let's unify the term `incremental Java/Groovy compilation`. However, the captalized `I` in `Incremental` is because `Incremental Java compilation` is a referenced section name - so I think here `I` is correct.